kedro/contrib/io/pyspark/spark_data_set.py

```diff
@@ -233,20 +240,18 @@ def _get_spark():
         return SparkSession.builder.getOrCreate()
 
     def _load(self) -> DataFrame:
-        load_path = self._fs_prefix + str(self._get_load_path())
+        load_path = _strip_dbfs_prefix(self._fs_prefix + str(self._get_load_path()))
 
         return self._get_spark().read.load(
             load_path, self._file_format, **self._load_args
         )
 
     def _save(self, data: DataFrame) -> None:
-        save_path = str(self._get_save_path())
-        data.write.save(
-            self._fs_prefix + save_path, self._file_format, **self._save_args
-        )
+        save_path = _strip_dbfs_prefix(self._fs_prefix + str(self._get_save_path()))
```

Comment:
How come we need `_fs_prefix` this time but not previously?

Reply:
In the past, `_fs_prefix` was added in the `save` call; i.e.

```
        data.write.save(
            self._fs_prefix + save_path, self._file_format, **self._save_args	
        )
```

I moved it up into the `save_path` definition, to make it consistent with how `_fs_prefix` is being added in the `load_path` definition (and not in the `load` call).